python/python/bystro/api/cli.py

```diff
@@ -0,0 +1,330 @@
+
+    if print_result:
+        if job_id:
+            print(f"\nFetching job with id:\t{job_id}")
+        else:
+            print(f"\nFetching jobs of type:\t{job_type}")
+
+    response = requests.get(url, headers=auth_header)
+
+    if response.status_code != 200:
+        raise RuntimeError(
+            f"Fetching jobs failed with response status: {response.status_code}. Error: {response.text}"
+        )
+
+    jobs = response.json()
+
+    if job_id:
+        # As mongo doesn't support keys with '.' in them, we need to store the config as a string
+        jobs["config"] = json.loads(jobs["config"])
+
+    if print_result:
+        print("\nJob(s) fetched successfully: \n")
+        print(json.dumps(jobs, indent=4))
+        print("\n")
+
+    return jobs
+
+
+def create_job(args: argparse.Namespace, print_result=True):
+    url, auth_header, _ = authenticate(args)
+    url = url + "/api/jobs/upload/"
+
+    payload = {
+        "job": json.dumps(
+            {
+                "assembly": args.assembly,
+                "options": {"index": args.index},
+            }
+        )
+    }
+
+    files = []
+    for file in args.files:
+        files.append(
+            (
+                "file",
+                (os.path.basename(file), open(file, "rb"), "application/octet-stream"), # noqa: SIM115
+            )
+        )
+
+    if print_result:
+        print(f"\nCreating jobs for files: {','.join(map(lambda x: x[1][0], files))}\n")
+
+    response = requests.post(url, headers=auth_header, data=payload, files=files)
+
+    if response.status_code != 200:
+        raise RuntimeError(
+            f"Job creation failed with response status: {response.status_code}.\
+                Error: \n{response.text}\n"
+        )
+
+    if print_result:
+        print("\nJob creation successful:\n")
+        print(json.dumps(response.json(), indent=4))
+        print("\n")
+
+    return response.json()
+
+
+def get_user(args, print_result=True):
+    if print_result:
+        print("\n\nFetching user profile\n")
+
+    res = authenticate(args)
+
+    url, auth_header, email = res
+
+    response = requests.request("GET", url + "/api/user/me", headers=auth_header)
+
+    if response.status_code != 200:
+        raise RuntimeError(
+            f"Fetching profile failed with response status: {response.status_code}.\
+                Error: \n{response.text}\n"
+        )
+
+    if print_result:
+        print(f"\nFetched Profile for email {email}\n")
+        print(json.dumps(response.json(), indent=4))
+        print("\n")
+
+    return response.json()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog="bystro-api", description="Bystro CLI tool for making API calls."
+    )
+    subparsers = parser.add_subparsers(title="commands")
+
+    # Adding the user sub-command
+    login_parser = subparsers.add_parser("login", help="Login to the system")
+    login_parser.add_argument("--host", required=True, help="Host of the system")
```

Comment:
we should probably document the expected format of the host, like the required `https://` prefix